File: frontend/src/app/features/dashboard/rooms/dtos/room-request.dto.ts
import {BaseResponseDto} from '../../../../core/dtos/base-response.dto';
import {RoomType} from '../enums/room.enum';

export interface RoomResponseDto extends BaseResponseDto {
  code: string;
  name: string;
  floor: number;
  building: number;
  capacity: number;
  area: number;
  type: RoomType;
}

File: backend/src/main/java/io/hb3nce04/keykeeperapp/feature/room/model/dto/RoomRequestDto.java
package io.hb3nce04.keykeeperapp.feature.room.model.dto;

import io.hb3nce04.keykeeperapp.feature.room.model.enums.RoomType;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class RoomRequestDto {
    @NotEmpty(message = "Kitöltése kötelező")
    private String code;

    @NotEmpty(message = "Kitöltése kötelező")
    private String name;

    private String floor;

    private String building;

    private Integer capacity;

    private Integer area;

    @NotNull(message = "Kitöltése kötelező")
    private RoomType type;
}
